.github/workflows/continous-integration-docker.yml

```diff
@@ -75,96 +76,45 @@ on:
-          echo "STATE_VERSION: $LOCAL_STATE_VERSION"
-
-          LWD_TIP_DISK=$(gcloud compute images list --filter="status=READY AND name~lwd-cache-.+-[0-9a-f]+-v${LOCAL_STATE_VERSION}-${NETWORK}-tip" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
-          if [[ -z "$LWD_TIP_DISK" ]]; then
-              echo "No TIP disk found for LWD"
-              echo "lwd_tip_disk=${{ toJSON(false) }}" >> "$GITHUB_OUTPUT"
-          else
-              echo "Disk: $LWD_TIP_DISK"
-              echo "lwd_tip_disk=${{ toJSON(true) }}" >> "$GITHUB_OUTPUT"
-          fi
-
-          ZEBRA_TIP_DISK=$(gcloud compute images list --filter="status=READY AND name~zebrad-cache-.+-[0-9a-f]+-v${LOCAL_STATE_VERSION}-${NETWORK}-tip" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
-          if [[ -z "$ZEBRA_TIP_DISK" ]]; then
-              echo "No TIP disk found for ZEBRA"
-              echo "zebra_tip_disk=${{ toJSON(false) }}" >> "$GITHUB_OUTPUT"
-          else
-              echo "Disk: $ZEBRA_TIP_DISK"
-              echo "zebra_tip_disk=${{ toJSON(true) }}" >> "$GITHUB_OUTPUT"
-          fi
-
-          ZEBRA_CHECKPOINT_DISK=$(gcloud compute images list --filter="status=READY AND name~zebrad-cache-.+-[0-9a-f]+-v${LOCAL_STATE_VERSION}-${NETWORK}-checkpoint" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
-          if [[ -z "$ZEBRA_CHECKPOINT_DISK" ]]; then
-              echo "No CHECKPOINT found for ZEBRA"
-              echo "zebra_checkpoint_disk=${{ toJSON(false) }}" >> "$GITHUB_OUTPUT"
-          else
-              echo "Disk: $ZEBRA_CHECKPOINT_DISK"
-              echo "zebra_checkpoint_disk=${{ toJSON(true) }}" >> "$GITHUB_OUTPUT"
-          fi
+  # Check if the cached state disks used by the tests are available for testnet.
+  #
+  # The outputs for this job have the same names as the workflow outputs in find-cached-disks.yml
+  # Some outputs are ignored, because we don't run those jobs on testnet.
+  get-available-disks-testnet:
+    name: Check if cached state disks exist for testnet
+    uses: ./.github/workflows/find-cached-disks.yml
+    with:
+      network: 'Testnet'
 
+  # Build the docker image used by the tests.
+  #
+  # The default network in the Zebra config in the image is mainnet, unless a manually triggered
+  # workflow or repository variable is configured differently. Testnet jobs change that config to
+  # testnet when running the image.
   build:
     name: Build CI Docker
     uses: ./.github/workflows/build-docker-image.yml
     with:
       dockerfile_path: ./docker/Dockerfile
       dockerfile_target: tests
       image_name: ${{ vars.CI_IMAGE_NAME }}
-      # The default network in the image is based on the default network variable in GitHub
-      network: ${{ vars.ZCASH_NETWORK }}
+      network: ${{ inputs.network || vars.ZCASH_NETWORK }}
       checkpoint_sync: true
```

Comment:
Please ignore everything above this line during reviews, that's getting reviewed in PR #6576.